File: app/src/main/java/app/kamy/saatApp/domain/model/OptionalWorshipHabit.kt
package app.kamy.saatApp.domain.model

enum class OptionalWorshipHabit(val prefKey: String) {
    QIYAMUL_LAIL("qiyam"),
    MONDAY_THURSDAY_FAST("mon_thu_fast"),
    AYYAMUL_BIDH_SAHUR("ayyamul_bidh_sahur"),
    DHIKR_MORNING("dhikr_morning"),
    DHIKR_EVENING("dhikr_evening"),
    READ_QURAN("read_quran"),
    DAILY_CHARITY("daily_charity"),
    DHUHA("dhuha"),
    RAWATIB("rawatib");

    companion object {
        val ALL = entries.toList()
    }
}

File: iosApp/Saat/Domain/Enums/OptionalWorshipHabit.swift
//
//  OptionalWorshipHabit.swift
//  Saat
//
//  Created by Elmee on 25/06/2026.
//  Copyright © 2026 Elmee. All rights reserved.
//

import Foundation

public enum OptionalWorshipHabit: String, CaseIterable, Identifiable, Sendable {
    case qiyamulLail = "QIYAMUL_LAIL"
    case mondayThursdayFast = "MONDAY_THURSDAY_FAST"
    case ayyamulBidhSahur = "AYYAMUL_BIDH_SAHUR"

    public var id: String { rawValue }

    public var prefKey: String {
        switch self {
        case .qiyamulLail: return "qiyam"
        case .mondayThursdayFast: return "mon_thu_fast"
        case .ayyamulBidhSahur: return "ayyamul_bidh_sahur"
        }
    }
}
